File: artifacts/api-server/Models/Exceptions.cs
namespace TimesheetApi.Models;

public class InvalidTransitionException : Exception
{
    public string CurrentStatus { get; }

    public InvalidTransitionException(string currentStatus, string attempted)
        : base($"Cannot perform '{attempted}' on a timesheet in '{currentStatus}' status")
    {
        CurrentStatus = currentStatus;
    }
}

File: artifacts/api-server/src/lib/exceptions.js
class InvalidTransitionException extends Error {
  constructor(currentStatus, attempted) {
    super(`Cannot perform '${attempted}' on a timesheet in '${currentStatus}' status`);
    this.name = "InvalidTransitionException";
    this.currentStatus = currentStatus;
  }
}

module.exports = { InvalidTransitionException };
